open_feature/open_feature_client.py

```diff
@@ -152,27 +166,65 @@ def evaluate_flag_details(
         evaluation_context: EvaluationContext = None,
         flag_evaluation_options: typing.Any = None,
     ) -> FlagEvaluationDetails:
+        """
+        Evaluate the flag requested by the user from the clients provider.
+
+        :param flag_type: the type of the flag being returned
+        :param key: the string key of the selected flag
+        :param default_value: backup value returned if no result found by the provider
+        :param evaluation_context: Information for the purposes of flag evaluation
+        :param flag_evaluation_options: Additional flag evaluation information
+        :return: a FlagEvaluationDetails object with the fully evaluated flag from a
+        provider
+        """
+
         if evaluation_context is None:
             evaluation_context = EvaluationContext()
+
+        hook_context = HookContext(
+            flag_key=key,
+            flag_type=flag_type,
+            default_value=default_value,
+            evaluation_context=evaluation_context,
+            client_metadata=None,
+            provider_metadata=None,
+        )
+        merged_hooks = []
+
         try:
-            # The merge below will eventually pull in Evaluation contexts from Hooks
-            invocation_context = EvaluationContext()
+            invocation_context = before_hooks(
+                flag_type, hook_context, merged_hooks, None
+            )
             invocation_context.merge(ctx2=evaluation_context)
-            return self.create_provider_evaluation(
+
+            # merge of: API.context, client.context, invocation.context
+            merged_context = (
+                get_evaluation_context().merge(self.context).merge(invocation_context)
+            )
+
+            flag_evaluation = self.create_provider_evaluation(
                 flag_type,
                 key,
                 default_value,
-                invocation_context,
+                merged_context,
                 flag_evaluation_options,
             )
+
+            after_hooks(type, hook_context, flag_evaluation, merged_hooks, None)
+
+            return flag_evaluation
         except OpenFeatureError as ofe:
+            error_hooks(flag_type, hook_context, merged_hooks, None)
```

Comment:
It looks like we're missing the `exception` argument from the function call here?

The definition of the function from the `hook_support` module:

```
def error_hooks(
    flag_type: FlagType,
    hook_context: HookContext,
    exception: Exception,
    hooks: typing.List[Hook],
    hints: dict,
)
```